File: src/main/java/com/example/gui/GuiHandler.java
package com.example.gui;

import com.example.inventory.ContainerChargeableDirt;
import com.example.inventory.ContainerChargeableDirt2;
import com.example.tileentity.TileEntityChargeableDirt;
import com.example.tileentity.TileEntityChargeableDirt2;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.tileentity.TileEntity;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;
import net.minecraftforge.fml.common.network.IGuiHandler;

public class GuiHandler implements IGuiHandler {
    public static final int CHARGEABLE_DIRT_GUI = 0;
    public static final int CHARGEABLE_DIRT2_GUI = 1;

    @Override
    public Object getServerGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        TileEntity te = world.getTileEntity(new BlockPos(x, y, z));

        if (ID == CHARGEABLE_DIRT_GUI) {
            if (te instanceof TileEntityChargeableDirt) {
                return new ContainerChargeableDirt(player, (TileEntityChargeableDirt) te);
            }
        } else if (ID == CHARGEABLE_DIRT2_GUI) {
            if (te instanceof TileEntityChargeableDirt2) {
                return new ContainerChargeableDirt2(player, (TileEntityChargeableDirt2) te);
            }
        }
        return null;
    }

    @Override
    public Object getClientGuiElement(int ID, EntityPlayer player, World world, int x, int y, int z) {
        TileEntity te = world.getTileEntity(new BlockPos(x, y, z));

        if (ID == CHARGEABLE_DIRT_GUI) {
            if (te instanceof TileEntityChargeableDirt) {
                return new GuiChargeableDirt(player.inventory, (TileEntityChargeableDirt) te);
            }
        } else if (ID == CHARGEABLE_DIRT2_GUI) {
            if (te instanceof TileEntityChargeableDirt2) {
                return new GuiChargeableDirt2(player.inventory, (TileEntityChargeableDirt2) te);
            }
        }
        return null;
    }
}

File: src/main/kotlin/wangyu/gui/GuiHandler.kt
package wangyu.gui

import wangyu.inventory.ContainerChargeableDirt
import wangyu.inventory.ContainerChargeableDirt2
import wangyu.tileentity.TileEntityChargeableDirt
import wangyu.tileentity.TileEntityChargeableDirt2
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.tileentity.TileEntity
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World
import net.minecraftforge.fml.common.network.IGuiHandler
import wangyu.ui.ExampleChargeableDirtGui

class GuiHandler : IGuiHandler {
    companion object {
        const val CHARGEABLE_DIRT_GUI = 0
        const val CHARGEABLE_DIRT2_GUI = 1
    }

    override fun getServerGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): Any? {
        val te = world.getTileEntity(BlockPos(x, y, z))

        return when (ID) {
            CHARGEABLE_DIRT_GUI -> {
                if (te is TileEntityChargeableDirt) {
                    ContainerChargeableDirt(player, te)
                } else null
            }
            CHARGEABLE_DIRT2_GUI -> {
                if (te is TileEntityChargeableDirt2) {
                    ContainerChargeableDirt2(player, te)
                } else null
            }
            else -> null
        }
    }

    override fun getClientGuiElement(ID: Int, player: EntityPlayer, world: World, x: Int, y: Int, z: Int): Any? {
        val te = world.getTileEntity(BlockPos(x, y, z))

        return when (ID) {
            CHARGEABLE_DIRT_GUI -> {
                if (te is TileEntityChargeableDirt) {
                    GuiChargeableDirt(player.inventory, te)
                } else null
            }
            CHARGEABLE_DIRT2_GUI -> {
                if (te is TileEntityChargeableDirt2) {
                    ExampleChargeableDirtGui(player.inventory, te)
                } else null
            }
            else -> null
        }
    }
}